Adding contributing_template.md for teams to copy to their repos

yhuiYH opened the pull request:
Adding contributing_template.md for teams to copy to their repos

TEMPLATE INSTRUCTIONS FOR LIBRARY OWNERS:
Step 1: Make a copy of this template into a branch for your repository. Rename template file to CONTRIBUTING.md and ensure it is in the root of your repo or in .github/CONTRIBUTING.md
Step 2: Rename all instances of "LIBRARY_NAME" above to the name of your library.  Search/find/replace.
Step 3: Read the Motivation section below to get a sense for why we need to publish a CONTRIBUTING.md
Step 4: Review all 10 instruction sections below [PART1-PART10] with your team to come up with your CONTRIBUTING.md instructions for the community.  5 are optional ones.
Step 5: Ensure a Codeowners file is included in your repository with Maintainers/reviewers specified.
Step 6: Remove all HTML comments/Instructions/MIOpen examples from this template to make it release-ready.
Step 7: Commit and push this file to your public repo.

yhuiYH committed:
Adding contributing_template.md for teams to copy to their repos

yhuiYH committed:
revising with tech writer version

yhuiYH committed:
minor grammar/typos

yhuiYH committed:
move contributing_template.md into docs/templates

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+Step 2: Rename all instances of "LIBRARY_NAME" above to the name of your library.  Search/find/replace.
+Step 3: Read the Motivation section below to get a sense for why we need to publish a CONTRIBUTING.md
+Step 4: Review all 10 instruction sections below [PART1-PART10] with your team to come up with your CONTRIBUTING.md instructions for the community.  5 are optional ones.
+Step 5: Ensure a Codeowners file is included in your repository with Maintainers/reviewers specified.
```
Change to CODEOWNERS to match filename?
Do you want to include a link to [GitHub Documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) for more information regarding CODEOWNERS?

dgaliffiAMD replied:
Maintainers --> maintainers
Fixing casing of first character.

yhuiYH replied:
fixed both.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+
+MIOpen Example:
+* Please follow the generic MIOpen PR review process, as defined here.
+* Reviewers are listed in the Codeowners file
```
Change to CODEOWNERS to match filename?

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+### Process ###
+
+<!-- [PART8] Instructions:
+   * Outline the PR review process, most importantly, ensure list of reviewers are updated in Codeowners file 
```
Change to CODEOWNERS to match filename?

yhuiYH replied:
fixed

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+could obtain the detailed information.
+
+MIOpen Example:
+1. [MIOpen Wiki](https://github.com/ROCmSoftwarePlatform/MIOpen/wiki)
```
Fixing link to Wiki: https://github.com/**ROCm**/MIOpen/wiki

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+
+MIOpen Example:
+1. [MIOpen Wiki](https://github.com/ROCmSoftwarePlatform/MIOpen/wiki)
+1. [MIOpen Readme](https://github.com/ROCmSoftwarePlatform/MIOpen#readme)
```
Fixing link to README: https://github.com/**ROCm**/MIOpen#README

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+Step 3: Read the Motivation section below to get a sense for why we need to publish a CONTRIBUTING.md
+Step 4: Review all 10 instruction sections below [PART1-PART10] with your team to come up with your CONTRIBUTING.md instructions for the community.  5 are optional ones.
+Step 5: Ensure a Codeowners file is included in your repository with Maintainers/reviewers specified.
+Step 6: Remove all HTML comments/Instructions/MIOpen examples from this template to make it release-ready.
```
Instructions --> instructions
Fixing casing on the first character.

yhuiYH replied:
fixed the typo

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+their very first pull request influences not only their workflow for subsequent
+pull requests but also their recommendations for other developers.
+
+AMD's recent engagement with its strategic partner in ROCm has revealed to us 
```
Partner**s** (pluralize partner)? Or, "a strategic partner", rather than "its strategic partner".

yhuiYH replied:
changed it to "a strategic partner".  This text is taken from Tejash's PR.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+
+For illustration, AMD's strategic partner wasn't aware about the acceptance criteria 
+of MIOpen and thereby, assumed that any deep learning kernels could be contributed 
+into MIOpen. In the same vein, they were unaware about the composable kernel library 
```
"composable_kernel"? To match the name of the repository.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+code duplication and code maintenance if no gain is found. Contributors should be guided about 
+any performance benchmarking mechanism including baseline comparison. 
+
+Additionally, to encourage contributors, we should point out that the "Issue" tab, as defined below, 
```
"as defined below": Are you referring to the "Issue Discussion" section **above**?

yhuiYH replied:
removed this "below"/"above" part.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+
+Additionally, to encourage contributors, we should point out that the "Issue" tab, as defined below, 
+could be leveraged by them to discuss with the community members, features they would like to 
+bring in or the purported perf gains it would bring, before raising pull requests.
```
This sentence is difficult to follow. Can we break it up?

yhuiYH replied:
Yeah, reworded this whole section from Tejash... hopefully I got across his points.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+
+<!-- [PART8] Instructions:
+   * Outline the PR review process, most importantly, ensure list of reviewers are updated in Codeowners file 
+   * List out code formatting steps.
```
Is this different than the "Coding Style" section? 
From the example, do you mean: "How to format your commit message" or "How to run your linter".

yhuiYH replied:
No, not the commit message.  He didn't have a Coding Style section before, I added it.  So it might be duplicate.  I added some text to say that, as part of the PR process, just ensure the contributor has followed the Coding Style section.

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
+     - Until ROCm5.6,  fp16, bfloat16, fp32
+     - Starting ROCm5.7, f8 
++ Introduce the host-side API, if required
+E.g. For LayerNorm, https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/include/miopen/miopen.h
```
https://github.com/**ROCm**/MIOpen/blob/develop/include/miopen/miopen.h

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
++ Introduce the host-side API, if required
+E.g. For LayerNorm, https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/include/miopen/miopen.h
++ Add unit tests in gtest format
+Dir: https://github.com/ROCmSoftwarePlatform/MIOpen/tree/5c8118ee4339c2aae55f20241440af207e59d2a7/test/gtest
```
https://github.com/**ROCm**/MIOpen/tree/5c8118ee4339c2aae55f20241440af207e59d2a7/test/gtest

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
++ Add unit tests in gtest format
+Dir: https://github.com/ROCmSoftwarePlatform/MIOpen/tree/5c8118ee4339c2aae55f20241440af207e59d2a7/test/gtest
++ Include the relevant code in driver to measure per-kernel performance
+https://github.com/ROCmSoftwarePlatform/MIOpen/tree/master/driver
```
https://github.com/**ROCm**/MIOpen/tree/master/driver

dgaliffiAMD commented on docs/templates/contributing_template.md:
```diff
@@ -0,0 +1,281 @@
++ Post the performance # against the baseline kernel.
++ For new kernels in MIOpen, e.g. LayerNorm 
+   - obtain the baseline from ROCm/Pytorch build: https://github.com/ROCmSoftwarePlatform/pytorch/blob/main/aten/src
+/ATen/native/cuda/layer_norm_kernel.cu
```
https://github.com/**ROCm**/pytorch/blob/main/aten/src
/ATen/native/cuda/layer_norm_kernel.cu

yhuiYH replied:
fixed

dgaliffiAMD committed:
Fixed some linter warnings

- Removed whitespace at the end of each line.
- Ensure the declarations of headings are consistant. ie., ensure opening and closing tags.

yhuiYH committed:
incorporating text changes and suggestions, and update URLs to ROCm org
